File: src/main/scala/org/ximo/finkinaction/scala/dataset/DataSetFirstNApp.scala
package org.ximo.finkinaction.scala.dataset
import org.apache.flink.api.common.operators.Order
import org.apache.flink.api.scala.{ExecutionEnvironment, _}

/**
  *
  *
  * @author xikl
  * @date 2019/8/5
  */
object DataSetFirstNApp {


  def main(args: Array[String]): Unit = {
    val env = ExecutionEnvironment.getExecutionEnvironment

    val info = List(
      (1, "hadoop"),
      (1, "scala"),
      (1, "java"),
      (2, "python"),
      (2, "vue"),
      (3, "flutter"),
      (4, "flink")
    )
    val data = env.fromCollection(info)

    // (1,hadoop)
    //(1,scala)
    //(1,java)
    data.first(3).print()

    // (3,flutter)
    //(1,hadoop)
    //(1,scala)
    //(4,flink)
    //(2,python)
    //(2,vue)
    data groupBy 0 first 2 print()

    //(3,flutter)
    //(1,hadoop)
    //(1,java)
    //(4,flink)
    //(2,python)
    //(2,vue)
    data groupBy 0 sortGroup(1, Order.ASCENDING) first 2 print()

  }

}

File: src/main/java/org/ximo/finkinaction/java/dataset/DataSetFirstNApp.java
package org.ximo.finkinaction.java.dataset;

import org.apache.flink.api.common.operators.Order;
import org.apache.flink.api.java.ExecutionEnvironment;
import org.apache.flink.api.java.tuple.Tuple2;

import java.util.List;
import java.util.stream.Stream;

import static java.util.stream.Collectors.toList;

/**
 * @author xikl
 * @date 2019/8/5
 */
public class DataSetFirstNApp {

    public static void main(String[] args) throws Exception {
        ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
        List<Tuple2<Integer, String>> data = Stream.of(
                new Tuple2<>(1, "scala"),
                new Tuple2<>(1, "java"),
                new Tuple2<>(1, "python"),
                new Tuple2<>(2, "flink"),
                new Tuple2<>(2, "saprk"),
                new Tuple2<>(3, "netty"),
                new Tuple2<>(4, "vue")
                ).collect(toList());

        env.fromCollection(data)
                .groupBy(0)
                .first(3)
                .print();

        env.fromCollection(data)
                .groupBy(0)
                .sortGroup(1, Order.ASCENDING)
                .first(3)
                .print();
    }
}
